src/io/obf.rs

```diff
@@ -131,7 +131,11 @@ where
     let identity_d_plus_1 = M::identity(params.as_ref(), d + 1, None);
     let level_width = obf_params.level_width; // number of bits to be inserted at each level
     assert_eq!(obf_params.input_size % level_width, 0);
-    assert_eq!(dim % level_width, 0);
+    assert!(level_width <= dim); // otherwise we need >1 polynomial to insert the bits for each level
+    if obf_params.input_size > dim {
+        assert_eq!(dim % level_width, 0); // otherwise we get to a point in which the inserted bits
```

Comment:
I think we can just assume that `level_width < dim`.
Since `dim` is usually larger than 1024 but `2^1024` is too large, `level_width >= dim` should not hold in practice.